server/datastore/mysql/labels.go

```diff
@@ -415,6 +415,11 @@ func (d *Datastore) ListHostsInLabel(filter fleet.TeamFilter, lid uint, opt flee
 	sql, params = filterHostsByStatus(sql, opt, params)
 	sql, params = searchLike(sql, params, opt.MatchQuery, hostSearchColumns...)
 
+	if opt.TeamFilter != nil {
+		sql += ` AND h.team_id = ?`
+		params = append(params, *opt.TeamFilter)
+	}
```

Comment:
This should probably go into a `filterHostsByTeam(sql, opt, params)` function and follow the pattern as `filterHostsByStatus`